File: LeetCode/cpp/965.cpp
class Solution {
public:
    bool isUnivalTree(TreeNode *root) {
        if (!root)
            return true;
        return Traverse(root, root->val);
    }

    bool Traverse(TreeNode *node, const int &value) {
        if (!node)
            return true;
        if (node->val != value)
            return false;
        return Traverse(node->left, node->val) && Traverse(node->right, node->val);
    }
};

File: LeetCode/python3/965.py
class Solution:
    value = 0

    def isUnivalTree(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if not root:
            return True
        return self.CompareValue(root, root.val)

    def CompareValue(self, root, value):
        if not root:
            return True
        if root.val != value:
            return False
        return self.CompareValue(root.left, value) and self.CompareValue(root.right, value)